back/flag-service/src/flag/user/spec/UserController.spec.ts

```diff
@@ -0,0 +1,340 @@
+              throw new IncorrectPasswordError();
+            });
+          res = userController.changePassword('anoldsecret135', 'thisisasecret123', 'thisisasecret123');
+        });
+        it('calls changePassword from service', () => {
+          expect(changePasswordSpy).toBeCalledTimes(1);
+        });
+        it('throws a BadRequestException when service throws a IncorrectPasswordError', async () => {
+          await expect(res).rejects.toThrow(BadRequestException);
+        });
+      });
+
+      describe('bad field values', () => {
+        function testValues(name: string, currentPassword: string, newPassword: string, newPasswordConfirmation: string) {
+          describe(name, () => {
+            let changePasswordSpy;
+            let res;
+            beforeAll(async () => {
+              changePasswordSpy = jest
+                .spyOn(userService, 'changePassword')
+                .mockReturnValue({ fake: true } as any);
+              res = userController.changePassword(currentPassword, newPassword, newPasswordConfirmation);
+            });
+            it('doesn\'t call register from service', () => {
+              expect(changePasswordSpy).not.toBeCalled();
+            });
+            it('throws a BadRequestException', async () => {
+              await expect(res).rejects.toThrow(BadRequestException);
+            });
+          });
+        }
+
+        describe('currentPassword', () => {
+          testValues('too short', 'abc12', 'thisisasecret123', 'thisisasecret123');
+          testValues('no number', 'thisisasecret', 'thisisasecret123', 'thisisasecret123');
+          testValues('no letter', '1239009384657493', 'thisisasecret123', 'thisisasecret123');
+        });
+
+        describe('newPassword', () => {
+          testValues('too short', 'thisisasecret123', 'abc12', 'abc12');
+          testValues('no number', 'thisisasecret123', 'thisisasecret', 'thisisasecret');
+          testValues('no letter', 'thisisasecret123', '1239009384657493', '1239009384657493');
+        });
+
+        describe('newPasswordConfirmation', () => {
+          testValues('incorrect', 'thisisasecret123', 'thisisasecret123', 'thisisasecret123 incorrect');
+        });
+      });
+    });
+  });
+
+  describe('changeNickname', () => {
+    describe('success', () => {
+      let changeNicknameSpy;
+      let res;
+      beforeAll(async () => {
+        changeNicknameSpy = jest
+          .spyOn(userService, 'changeNickname')
+          .mockReturnValue({ fake: true } as any);
+        res = await userController.changeNickname('thisisasecret123', 'jane2');
+      });
+      it('calls changeNickname from service', () => {
+        expect(changeNicknameSpy).toBeCalledTimes(1);
+      });
+      it('returns changeNickname return value', () => {
+        expect(res).toStrictEqual({ nickname: 'jane2' });
+      });
+    });
+    describe('failure', () => {
+      for (const error of [
+        new IncorrectPasswordError(),
```

Comment:
I don't think that IncorrectPassword can be thrown when changing the nickname

Reply:
This will be solved with https://github.com/Sorikairox/micronation/pull/3#discussion_r674725620